velox/functions/lib/DateTimeFormatter.h

```diff
@@ -191,5 +200,58 @@ std::shared_ptr<DateTimeFormatter> buildMysqlDateTimeFormatter(
 
 std::shared_ptr<DateTimeFormatter> buildJodaDateTimeFormatter(
     const std::string_view& format);
+namespace {
+inline constexpr int64_t kSecondsInDay = 86'400;
+inline constexpr int64_t kDaysInWeek = 7;
+
+FOLLY_ALWAYS_INLINE const date::time_zone* getTimeZoneFromConfig(
+    const QueryConfig& config) {
+  if (config.adjustTimestampToTimezone()) {
+    auto sessionTzName = config.sessionTimezone();
+    if (!sessionTzName.empty()) {
+      return date::locate_zone(sessionTzName);
+    }
+  }
+  return nullptr;
+}
+
+FOLLY_ALWAYS_INLINE int64_t
+getSeconds(Timestamp timestamp, const date::time_zone* timeZone) {
+  if (timeZone != nullptr) {
+    timestamp.toTimezone(*timeZone);
+    return timestamp.getSeconds();
+  } else {
+    return timestamp.getSeconds();
+  }
+}
+
+FOLLY_ALWAYS_INLINE
+std::tm getDateTime(Timestamp timestamp, const date::time_zone* timeZone) {
+  int64_t seconds = getSeconds(timestamp, timeZone);
+  std::tm dateTime;
+  gmtime_r((const time_t*)&seconds, &dateTime);
+  return dateTime;
+}
+
+FOLLY_ALWAYS_INLINE
+std::tm getDateTime(Date date) {
+  int64_t seconds = date.days() * kSecondsInDay;
+  std::tm dateTime;
+  gmtime_r((const time_t*)&seconds, &dateTime);
+  return dateTime;
+}
+
+template <typename T>
+struct InitSessionTimezone {
```

Comment:
This struct as well as getDateTime functions should not be in the anonymous namespace.